src/query_router.rs

```diff
@@ -150,9 +151,13 @@ impl QueryRouter {
                     || self.pool_settings.sharding_key_regex.is_some()
                 {
                     // Check only the first block of bytes configured by the pool settings
-                    let len = message_cursor.get_i32() as usize;
                     let seg = cmp::min(len - 5, self.pool_settings.regex_search_limit);
```

Comment:
Add a guard either here or below where this is used to make sure `seg` is >0

Reply:
Will discuss further about how to handle this and have a follow up PR to resolve this